src/compose/config.py

```diff
@@ -0,0 +1,195 @@
+
+
+def is_reachable(url: str) -> str:
+    """
+    Check if a string is a reachable URL.
+
+    Parameters:
+        url: URL string.
+
+    Returns:
+        URL string.
+
+    Raises:
+        ValueError: if the URL is not reachable.
+    """
+    req = request.Request(url, method=HTTPMethod.HEAD)
+    debug('Checking if URL is reachable: {0}.'.format(url))
+    try:
+        with request.urlopen(req) as response:  # noqa: S310
+            if response.status != HTTPStatus.OK:
+                msg = 'URL is unreachable: {0}:\n↳ Status Code: {1}'
+                raise ValueError(msg.format(url, response.status))
+    except URLError as error:
+        raise ValueError('URL is unreachable: {0}:\n↳ {1}'.format(url, error))
+    return url
+
+
+ReachableURL = Annotated[str, AfterValidator(is_reachable)]
+
+
+class LinkConfig(BaseModel):
+    """Parses and validates link configuration."""
+
+    name: str
+    url: ReachableURL
+
+
+class NewsConfig(BaseModel):
+    """Parses and validates news configuration."""
+
+    title: str
+    date: date
+    image: Optional[ReachableURL] = None
+    content: Optional[str] = None  # noqa: WPS110
+
+
+class ProjectConfig(BaseModel):
+    """Loads, parses and validates project sources configuration."""
+
+    id: str
+    url: str
+    type: Optional[Literal['featured', 'regular']] = 'regular'
+    name: str
+    description: str
+    website: ReachableURL
+    licenses: list[str]
+    image: Optional[ReachableURL] = None
+    documentation: Optional[ReachableURL] = None
+    issues: Optional[ReachableURL] = None
+    latest_release: Optional[ReachableURL] = None
+    forum: Optional[ReachableURL] = None
+    links: Optional[list[LinkConfig]] = None
+    gallery: Optional[list[ReachableURL]] = None
+    categories: Optional[list[str]] = None
+    tags: Optional[list[str]] = None
+    news: Optional[list[NewsConfig]] = None
+
+    @classmethod
+    def from_url(cls, url: str, **kwargs):
+        """
+        Load project sources configuration from a URL.
+
+        Parameters:
+            url: git repository URL.
+            kwargs: id and type (optional).
+
+        Returns:
+            ProjectConfig object.
+        """
+        if urlparse(url).hostname == 'github.com':
+            return cls.from_github(url=url, **kwargs)
+        return cls.from_repo(url=url, **kwargs)
+
+    @classmethod
+    def from_repo(cls, url: str, **kwargs):
+        """
+        Load project sources configuration from a git repository.
+
+        Parameters:
+            url: git repository URL.
+            kwargs: id and type (optional).
+
+        Returns:
+            ProjectConfig object.
+
+        Raises:
+            ConfigError: if loading the configuration fails.
+        """
+        info('Loading .ohwr.yaml from {0} with git clone...'.format(url))
+        tmpdir = TemporaryDirectory().name
+        try:
+            subprocess.check_output(
+                'git clone --depth 1 {0} {1}'.format(url, tmpdir),
```

Comment:
I could not resist to point out once more the f'string, it seems much more readable to me.
```suggestion
                f'git clone --depth 1 {url} {tmpdir}'
```

Reply:
Indeed, I also find f strings very elegant but wemake-python-styleguide doesn't allow f strings - https://wemake-python-styleguide.readthedocs.io/en/latest/pages/usage/violations/consistency.html#wemake_python_styleguide.violations.consistency.FormattedStringViolation